src/runtime_src/core/common/api/xrt_bo.cpp

```diff
@@ -514,7 +516,7 @@ alloc(xclDeviceHandle dhdl, size_t sz, xrtBufferFlags flags, xrtMemoryGroup grp)
   switch (type) {
   case 0:
 #ifndef XRT_EDGE
-    if (is_nodma())
+    if (is_nodma(dhdl))
       return alloc_nodma(dhdl, sz, flags, grp);
```

Comment:
will alloc_nodma return error if user tries to alloc normal BO?

Reply:
@maxzhen What is a normal BO?  Do you mean userptr bo?

Reply:
OK, it's good if driver returns failure.